entities/src/main/java/org/odk/collect/entities/storage/InMemEntitiesRepository.kt

```diff
@@ -56,7 +56,7 @@ class InMemEntitiesRepository : EntitiesRepository {
                         EntitySchema.VERSION -> it.version.toString()
                         else -> it.properties.find { propertyName ->
                             propertyName.first == query.column
-                        }?.second
+                        }?.second ?: throw QueryException("No such column: ${query.column}")
```

Comment:
There should really be a test in `EntitiesRepositoryTest` to make sure that both implementations are consistent here. I'm going to be doing a quick pass over at #6614 anyway though so will tidy that up when moving tests from `getAllByProperty` to `query`.